### PR TITLE
Development

### PR DESCRIPTION
**Pull request by swengineermwi**
Initial steps

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  for (let j = 0; j < mList.length; j++) {
+    const book = mList[j];
+    stack(`<div id="${j + book.title}">
+        <p>${book.title}</p>
+        <p>${book.author}</p>
+        <button id="${j + book.title}" value="${j}" onclick="removeBook(this.id, this.value)">Remove</button>
+      </div>
+      <hr>`);
+  }
+}
```
Comment (carlos-ssh) [inline app.js]: When I try to save my books and refresh the page, I can only see my last book.

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @softwareengineermwi  and @sneeu-leeu, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._ 👇🏼

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- A function that cleans the inputs once that you add a new book. (Input maintains the last data on it.)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+function removeBook(id, p) {
+  const position = parseInt(p, 10);
+  list.splice(position, 1);
+  document.getElementById(id).remove();
+  localStorage.setItem('stuff', JSON.stringify({ arr: list }));
+}
+/* eslint-disable no-plusplus */
+function loadPrev() {
+  const mList = JSON.parse(localStorage.getItem('stuff')).arr;
```
Comment (teekaytech) [inline app.js]: - [ ] As mentioned by the last reviewer, your app does not retain books data when the page is reloaded. It looks like you're having some issues with your localStorage. I tried to run your application locally and I found the following error in my console: 

shot

Can this be the problem? You might want to check this line and see what's actually causing the error:
https://github.com/softwareengineermwi/lib/blob/aaf1b2ab2f1abb8b8759dcbfe01b05549f65e6a1/app.js#L49

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

**Great work on making the changes requested by the previous reviewer 👏** 

_You've done well in implementing most of the requested changes, but there are still some which aren't addressed yet._

__However, you need to make some required changes before moving on to the next assignment.__
> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it :blush: 

### **Required Changes :recycle:**
_**Please check my inline comment below:**_

### **Optional Changes** :zap: 
N/A

* * *
**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review unless it is requested otherwise.**

*Feel free to leave any questions or comments in the PR thread if something is not 100% clear.*

*REMINDER: As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).*

_**Please make the required changes and submit another review request when you are done.**_

* * *

Re-reviewed by: [**Taofeek Olalere**](https://github.com/teekaytech)
*Cheers and Happy Coding* :smiley:

File: app.js (hunk 2 above)
Comment (mricanho) [reply app.js]: - You are using arr without defining it first, as well as stuff, that's why the null

File: app.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  g('a').value = '';
+}
+function removeBook(id, p) {
+  const position = parseInt(p, 10);
+  if (fromLocalStorage) {
+    list.splice(position, 1);
+    document.getElementById(id).remove();
+    updateLocalStorage();
+  } else {
+    const locaList = JSON.parse(localStorage.getItem('stuff')).arr;
```
Comment (mricanho) [inline app.js]: - The same problem with arr

File: app.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+function addBook() {
+  const title = g('t').value;
+  const author = g('a').value;
+  stack(`<div id="${i + title}">
+      <p>${title}</p>
+      <p>${author}</p>
+      <button id="${i + title}" value="${i}" onclick="removeBook(this.id, this.value)">Remove</button>
+    </div>
+    <hr>`);
+  const locaList = JSON.parse(localStorage.getItem('stuff')).arr;
```
Comment (mricanho) [inline app.js]: - The same problem with arr, you are using it as a function, but it's not, if intend to push it into the list, you need to use push instead

**Review by mricanho — CHANGES_REQUESTED** [review]
Hi @sneeu-leeu @softwareengineermwi, 

Generally, it's not a bad attempt, for the next one improve the PR description, your commits are good and consistent as well as your gitflow.


 ## Required Changes ♻️

- [ ] The data it's not preserved in the local storage once you reload the page, and the remove button doesn't work. The main problem it's the definition of some variables on your JS code, this blocks the local storage action, you need to refactor your methods to achieve the desired behavior.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

pika sad
you can do it

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nidalaa — APPROVED** [review]

**Review by nidalaa — APPROVED** [review]
Hello  @softwareengineermwi and @sneeu-leeu !

_This is an additional review as a result of the second opinion request._

After checking all information provided by you and Code Reviewers engaged in this case, we decided that your project should be approved.

All code reviewers were checking your app by using the online service and while in 99% of cases it works exactly the same as in the case of the local env, they experienced the errors mentioned in the review. We double-checked those errors and as edge cases, they should not be a reason to not approve your project. However, I recommend you trying to make your code strong enough to work also with [githack](https://raw.githack.com/) in your free time. That could be a fun way to play with JavaScript 🔍 

We are really sorry that this issue was not solved earlier - it could have been done after the first review.  For that reason, I really want to ask you to bring up this kind of issue in PR comments. If you have left a message under the comment from the first reviewer saying that "everything works in my local env", the next reviewer could have seen that and react to that. Without such information, each reviewer was starting from the same place.

**Please fix the linter errors that have appeared since the last review and merge this PR.**

Your project will be automatically marked as complete in your Dashboard within 1 hour.

Happy coding! 


**https://media.giphy.com/media/LmNwrBhejkK9EFP504/giphy.gif**

**Review by Dipeshtwis — COMMENTED** [review]
# Status: Invalid ❌

Please fix the issue mentioned below and then request for code review

## Reason

- [ ] You have requested code review for another project in the same pull request. Please create a new branch, make the changes, and open a new PR for this project (Awesome books: refactor to use JavaScript classes).

### Please make sure to submit code review when you are done with the changes that was requested

**Don't worry you still have same number of requests left as previous.**


Cheers and Happy Coding :clap: :clap: :clap: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Dipesh Kumar_**

**Pull request merged**